deploy/crd/operatingsystemmanager.k8c.io_operatingsystemprofiles.yaml

```diff
@@ -174,7 +174,6 @@ spec:
                         name:
                           description: Name of the Container runtime
                           enum:
-                          - docker
```

Comment:
This requires ether a new apiVersion, as it's heavy backward incompatible or some deprecation / defaulting / whatever else means.

Just deleting it will make everyone angry.

Reply:
Ah, you are right. Although `docker` is not an acceptable value anymore. How about keeping this field as is and then rejecting this via webhooks?

Reply:
That will render already deployed OSPs broken. I'm not entirely sure but probably we should forcefully migrate it from docker to containerd (in the mutation WH) and then proceed.

Reply:
I think you are confusing it a bit. OSPs are supposed to hold configuration(read scripts) for the container runtime. For example, in OSM v1.0.0 we had default OSPs with configuration for [containerd](https://github.com/kubermatic/operating-system-manager/blob/release/v1.0/deploy/osps/default/osp-ubuntu.yaml#L95) and [docker](https://github.com/kubermatic/operating-system-manager/blob/release/v1.0/deploy/osps/default/osp-ubuntu.yaml#L131).

Which container runtime is going to be used for the generated user-data(OSC) is then determined by what was configured at OSM level via [flags](https://github.com/kubermatic/operating-system-manager/blob/main/cmd/osm-controller/main.go#L112). With OSM [v1.2.0](https://github.com/kubermatic/operating-system-manager/releases/tag/v1.2.0) we already dropped support for docker so even if someone specifies configs for `docker` in OSP it's no-op.

Reply:
So there is no possibility for "mutation" to migrate docker to containerd here. But I do agree that removing the field within the same API version is not acceptable either.

Reply:
I think OSM should just ignore or emit an event if it encounters a now-broken value in the CRD, rather than breaking the CRD. Breaking the CRD will just cause so much trouble down the line. And all this complexity with migration webhooks will just lead to drift between gitops state and cluster state and is IMHO not worth it.

Reply:
Yeah that's exactly what I was thinking. In general, we don't even need the event. You can't configure `container-runtime` to be `docker`. So all in all it's no-op. Maybe a comment on top that `docker` is no-op/not supported is enough.

Reply:
I have updated the PR to reflect what i'm thinking of doing, a proposal. The conversation is still on going so it's not the final decision.